### PR TITLE
Archive

### PR DESCRIPTION
**Pull request by dloney**
This addresses #28 and #27. The Windows exclude functionality wasn't intended to be functional within the model and this would eliminate that issue.

This upgrades the C++ dependencies to the 2017 standard to allow use of the filesystem class. Incorporating that class reduces the separate handling of file operations between Linux and Windows, simplifying the code stack by making file operations platform independent. 

Documentation is also forked for v17.12/v21.03.06 and a dedicated development documentation branch was created.

### CODE REVIEW

**Review by lbearup — APPROVED** [review]

File: src/Model.cpp [hunk 1]
```diff
@@ -54,6 +54,8 @@ Version History
 #include <mpi.h>
 #include <math.h>
 #include <string.h>
+#include <string>
```
Comment (julemai) [inline src/Model.cpp:57]: Does it need `<string.h>` AND `<string>`?

**Comment by julemai** [comment]
LGTM. I am guessing that you have tested that this all works on non-Windows systems, right?

File: src/Model.cpp (hunk 1 above)
Comment (dloney) [reply src/Model.cpp:57]: Yes. string.h is a c class for character arrays. string is the cpp version for its string classes.

**Comment by dloney** [comment]
I did test on Windows as well. Eliminate the different handling between the filesystems should help with maintainability.

**Review by julemai — APPROVED** [review]

**Pull request merged**